File: fastcampus/01.완전탐색/15649.py
import sys
sys.setrecursionlimit(10**8)
input = sys.stdin.readline

n, m = map(int, input().split())

def rec(n, results):
    if len(results) == m:
        print(' '.join(map(str, results)))
        return
    for i in range(1, n+1):
        if i not in results:
          results.append(i)
          rec(n, results)
          results.pop()

rec(n, [])

File: fastcampus/01.완전탐색/15649.js
let fs = require('fs');
// let input = fs.readFileSync('/dev/stdin').toString();
let input = fs.readFileSync('ex.txt').toString();

input = input.split(' ')

let [n, m] = input.map(el => parseInt(el))
let arr = [];
let result = [];

function rec(k){
  if(k === m){
    result.push(arr.join(' '))
    return
  }
  for(let i=1; i <= n; i ++){
    if(!arr.includes(i)){
      arr.push(i)
      rec(k+1)
      arr.pop()
    }
  }
}

rec(0)
console.log(result.join('\n'))
